_posts/blog/2021-04-21-jshell-accumulo-feature.md

```diff
@@ -0,0 +1,141 @@
+---
+Title: JShell Accumulo Feature
+Author: R. Dane Magbuhos
+Reviewers: Christopher Tubbs, Keith Turner
+---
+
+## Overview
+First introduced in Java 9+, [JShell][JShell Doc] is an interactive Read-Evaluate-Print-Loop (REPL) 
+Java tool that assess user's inputed declarations, statements, and expressions and outputs 
+the results. This tool provides a convenient way to test out and execute quick tasks with Accumulo
+in the terminal.
+
+## Major Features
+* During Accumulo build, produces a default JShell script called `jshell-init.jsh` 
+containing up-to-date [Accumulo Java APIs][public APIs] and [AccumuloClient][client] 
+build implementation
+
+* Startup JShell with default or custom JShell script 
+
+* Both JShell start up options automatically import all relevant Java APIs 
+
+## Booting Up JShell Accumulo
+After installing and configuring the [latest Accumulo update][accumulo-repo]
```

Comment:
Since this is a 2.1 feature, we could point to the downloads page. If we want to publish this blog post before 2.1 is released, then some of the wording could be tailored to refer to the "upcoming 2.1 release" for this feature, with perhaps a minor mention that they could build pre-release snapshot versions to try it out from the git repository if 2.1 isn't released by the time they read this.

Reply:
I believe we can post this blog before the 2.1 release. That way users can get a small glimpse of what 2.1 has to offer.

Reply:
I agree. However, we do need to be careful that we're not violating ASF policy regarding publication of non-released material: https://www.apache.org/legal/release-policy.html#publication

Specifically, we want to avoid directing users (non-devs for the project) to unreleased content.

Reply:
Quick question: Since JShell Accumulo has been merged with the main Accumulo repo, would it be safe to link the main Accumulo repo on the blog post?

Reply:
The git repository hosting Accumulo source code, and especially the branch where this feature is merged, contains unreleased content. See the definition of "release" at ASF at https://www.apache.org/legal/release-policy.html#release-definition

I advise not linking against it. We can link to the downloads page for "when it becomes available in 2.1", and we can make a reference to the unreleased code in the git repository, but we probably shouldn't link to it. Developers who are involved already understand how to access unreleased code, so don't need to be told, and users who aren't involved in the development shouldn't be directed to use any unreleased code.

So, I think one possible way to frame this is with the following elements:
1. Explanation of upcoming feature
2. If you're a developer who wants to get involved in testing, contact us (link to our contact-us page) or see our contributing guide (link to contributing docs)
3. Link to download page with some kind of statement like "look for this feature in an upcoming 2.1 release at our downloads page" or something along those lines.

Reply:
Ahhh I see. That makes more sense. I'll get rid of the repo link and will implement your suggested idea.